module/VuFind/src/VuFind/Export/Zotero/ZoteroOAuth.php

```diff
@@ -0,0 +1,217 @@
+<?php
+
+/**
+ * Zotero OAuth Connector
+ *
+ * PHP version 8
+ *
+ * Copyright (C) The National Library of Finland 2025.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Zotero
+ * @author   Ere Maijala <ere.maijala@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+
+declare(strict_types=1);
+
+namespace VuFind\Export\Zotero;
+
+use League\OAuth1\Client\Credentials\ClientCredentialsInterface;
+use League\OAuth1\Client\Credentials\CredentialsException;
+use League\OAuth1\Client\Credentials\TokenCredentials;
+use League\OAuth1\Client\Server\User;
+use League\OAuth1\Client\Signature\SignatureInterface;
+use VuFind\Http\GuzzleService;
+
+use function is_array;
+
+/**
+ * Zotero OAuth Connector
+ *
+ * Class for authenticating with Zotero using OAuth.
+ *
+ * @category VuFind
+ * @package  Zotero
+ * @author   Ere Maijala <ere.maijala@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+class ZoteroOAuth extends \League\OAuth1\Client\Server\Server
+{
+    /**
+     * Constructor.
+     *
+     * @param GuzzleService                    $guzzleService     Guzzle service
+     * @param string                           $serviceName       VuFind title
+     * @param ClientCredentialsInterface|array $clientCredentials Client credentials
+     * @param SignatureInterface               $signature         Signature
+     */
+    public function __construct(
+        protected GuzzleService $guzzleService,
+        protected string $serviceName,
+        $clientCredentials,
+        ?SignatureInterface $signature = null
+    ) {
+        parent::__construct($clientCredentials, $signature);
+    }
+
+    /**
+     * Get the OAuth URL for retrieving temporary credentials.
+     *
+     * @return string
+     */
+    public function urlTemporaryCredentials()
+    {
+        return 'https://www.zotero.org/oauth/request';
```

Comment:
Should we consider making the Zotero base URL a class property so it's easier to change/override if there are adjustments in the future?